client/data/newsletter-categories/use-newsletter-categories-query.tsx

```diff
@@ -0,0 +1,58 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import request from 'wpcom-proxy-request';
+
+type NewsletterCategoryQueryProps = {
+	blogId: number;
+};
+
+type NewsletterCategoryQueryResponse = {
+	newsletter_categories: NewsletterCategoryResponse[];
+};
+
+type NewsletterCategories = {
+	newsletterCategories: NewsletterCategory[];
+};
+
+type NewsletterCategoryResponse = {
+	term_id: number;
+	name: string;
+	description: string;
+	order: number;
+};
+
+type NewsletterCategory = {
+	termId: number;
+	name: string;
+	description: string;
+	order: number;
+};
```

Comment:
This is a more general type and could be moved to a types file. What do you think?

Reply:
Sure! I'll do it!